kubernetes/k8_workers.tf

```diff
@@ -16,8 +16,8 @@ resource "tls_cert_request" "nfs_tunnel_client_request" {
 resource "tls_locally_signed_cert" "nfs_tunnel_client_certificate" {
   count = local.params.kubernetes.workers.nfs_tunnel ? 1 : 0
   cert_request_pem   = tls_cert_request.nfs_tunnel_client_request.0.cert_request_pem
-  ca_private_key_pem = file("${path.module}/../shared/nfs-ca.key")
-  ca_cert_pem        = file("${path.module}/../shared/nfs-ca.crt")
+  ca_private_key_pem = fileexists("${path.module}/../shared/nfs-ca.key") ? file("${path.module}/../shared/nfs-ca.key") : ""
+  ca_cert_pem        = fileexists("${path.module}/../shared/nfs-ca.crt") ? file("${path.module}/../shared/nfs-ca.crt") : ""
```

Comment:
Right.

If the tunnel is disabled, this should not run and if the tunnel is enabled, then you need those files and it is best to have a failure right there, otherwise you will have issues.

I'd replace the check that the file exists with a check on the boolean `local.params.kubernetes.workers.nfs_tunnel` instead.

That should do the trick